web/src/test/auto/headless/engine/interfaces/prediction/predictionContext.spec.js

```diff
@@ -118,6 +118,65 @@ describe("PredictionContext", () => {
     assert.equal(suggestions.find((obj) => obj.transform.deleteLeft != 0).displayAs, 'apps');
   });
 
+  it('ignores outdated predictions', async function () {
+    const langProcessor = new LanguageProcessor(worker, new TranscriptionCache());
+    await langProcessor.loadModel(appleDummyModel);  // await:  must fully 'configure', load script into worker.
+
+    const kbdProcessor = new KeyboardProcessor(deviceSpec);
+    const predictiveContext = new PredictionContext(langProcessor, kbdProcessor);
+
+    let updateFake = sinon.fake();
+    predictiveContext.on('update', updateFake);
+
+    let mock = new Mock("appl", 4); // "appl|", with '|' as the caret position.
+    const initialMock = Mock.from(mock);
+    const promise = predictiveContext.setCurrentTarget(mock);
+
+    // Initial predictive state:  no suggestions.  context.initializeState() has not yet been called.
+    assert.equal(updateFake.callCount, 1);
+    assert.isEmpty(updateFake.firstCall.args[0]); // should have no suggestions. (if convenient for testing)
+
+    await promise;
+    let suggestions;
+
+    // Initialization results:  our first set of dummy suggestions.
+    assert.equal(updateFake.callCount, 2);
+    suggestions = updateFake.secondCall.args[0];
+    assert.deepEqual(suggestions.map((obj) => obj.displayAs), ['apple', 'apply', 'apples']);
+    assert.isNotOk(suggestions.find((obj) => obj.tag == 'keep'));
+    assert.isNotOk(suggestions.find((obj) => obj.transform.deleteLeft != 0));
+
+    const baseTranscription = mock.buildTranscriptionFrom(initialMock, null, true);
+
+    // Mocking:  corresponds to the second set of mocked predictions - round 2 of
+    // 'apple', 'apply', 'apples'.
+    const skippedPromise = langProcessor.predict(baseTranscription, kbdProcessor.layerId);
```

Comment:
In retrospect, I didn't end up using this value.  Not sure what the best way to test with it would be, so maybe I should just...

```suggestion
    langProcessor.predict(baseTranscription, kbdProcessor.layerId);
```

and disregard its returned value completely... which is what happens in production, anyway.